trlx/utils/__init__.py

```diff
@@ -46,6 +49,23 @@ def safe_mkdir(path: str):
     os.mkdir(path)
 
 
+def get_distributed_config(accelerator: Accelerator):
+    """
+    Return accelerator distributed config
+    """
+    ds_plugin = accelerator.state.deepspeed_plugin
+    accelerate_config = accelerator.state
+    return {
+        "mixed_precision": accelerate_config.mixed_precision,
+        "num_gpus": accelerate_config.num_processes,
+        "gradient_accumulation_steps": ds_plugin.gradient_accumulation_steps,
```

Comment:
I think if ds is not used then `ds_plugin` is None, giving attribute error here
https://github.com/huggingface/accelerate/blob/e4e5611e5d4270a846caf42cba3388e54b83f074/src/accelerate/state.py#L62

maybe some processing of `repr(accelerator.state)` (which is outputed in `accelerate env`) would be equivalent here

Reply:
I guess it's true in some cases a user may not be using deepspeed so a check should be performed. However with the accelerator.state object there are some items I don't care about (such as the local rank).